### PR TITLE
[processing] Tests for eliminate

### PR DESCRIPTION
**Pull request by bstroebl**
I changed poly.gml and reran the algs that use it as input
Eliminate does not depend on an editable input layer any more, therefore custom/polys.shp could be deleted

### CODE REVIEW

File: python/plugins/processing/algs/qgis/Eliminate.py [hunk 1]
```diff
@@ -108,8 +109,8 @@ def processAlgorithm(self, progress):
             comparison = self.comparisons[self.getParameterValue(self.COMPARISON)]
             comparisonvalue = self.getParameterValue(self.COMPARISONVALUE)
 
-            selectindex = inLayer.dataProvider().fieldNameIndex(attribute)
-            selectType = inLayer.dataProvider().fields()[selectindex].type()
+            selectindex = vector.resolveFieldIndex(processLayer, attribute)
+            selectType = processLayer.dataProvider().fields()[selectindex].type()
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/Eliminate.py]: In general it's better not to use the dataProvider(). `processLayer.dataProvider().fields()` -> `processLayer.fields()`

**Comment by m-kuhn** [comment]
Nice :+1:

File: python/plugins/processing/algs/qgis/Eliminate.py (hunk 1 above)
Comment (bstroebl) [inline python/plugins/processing/algs/qgis/Eliminate.py]: done with next commit

**Comment by bstroebl** [comment]
danke

**Pull request merged**